### PR TITLE
Refactor CIFAR Example Based on the Latest AdaptDL

### PR DESCRIPTION
**Pull request by hao-howard-zhang**
The old CIFAR example is based on an outdated version of AdaptDL. This PR revamped the example based on [the official AdaptDL CIFAR example](https://github.com/petuum/adaptdl/blob/master/examples/pytorch-cifar/main.py).

I have tested it on the cluster.

### CODE REVIEW

File: examples/trials/cifar-adaptdl/main.py [hunk 1]
```diff
@@ -124,21 +116,18 @@ def train(epoch):
         stats["total"] += targets.size(0)
         stats["correct"] += predicted.eq(targets).sum().item()
 
-        writer.add_scalar("Throughput/Gain", net.gain, epoch)
-        writer.add_scalar("Throughput/Global_Batchsize",
-                          trainloader.current_batch_size, epoch)
-
+    trainloader.to_tensorboard(writer, epoch, tag_prefix="AdaptDL/Data/")
```
Comment (ZeyaWang) [inline examples/trials/cifar-adaptdl/main.py:119]: Is "to_tensorboard" a new feature? I did not see it in other examples.

File: examples/trials/cifar-adaptdl/main.py [hunk 2]
```diff
@@ -155,22 +144,28 @@ def valid(epoch):
...
 tensorboard_dir = os.path.join(
     os.getenv("ADAPTDLCTL_TENSORBOARD_LOGDIR", "/adaptdl/tensorboard"),
     os.getenv("NNI_TRIAL_JOB_ID", "cifar-adaptdl")
 )
 if not os.path.exists(tensorboard_dir):
     os.makedirs(tensorboard_dir)
+
 with SummaryWriter(tensorboard_dir) as writer:
-    for epoch in et.remaining_epochs_until(args.epochs):
+    acc = 0
```
Comment (ZeyaWang) [inline examples/trials/cifar-adaptdl/main.py:164]: "acc=0" can be removed.

**Review by ZeyaWang — COMMENTED** [review]
The branch can be merged after all the reviews are resolved.

File: examples/trials/cifar-adaptdl/main.py (hunk 1 above)
Comment (hao-howard-zhang) [reply examples/trials/cifar-adaptdl/main.py:119]: I see it in the updated AdatptDL example: https://github.com/petuum/adaptdl/blob/master/examples/pytorch-cifar/main.py#L127

File: examples/trials/cifar-adaptdl/main.py (hunk 2 above)
Comment (hao-howard-zhang) [reply examples/trials/cifar-adaptdl/main.py:164]: This is more of the code style preference. In most other languages, we must declare a variable first. So this line is to conform with this tradition. Besides, if we don't declare `acc=0`, a lint error will be raised in my IDE ("possible unbound variables").

**Review by ZeyaWang — APPROVED** [review]

**Pull request merged**